src/icons/fileIcons.ts

```diff
@@ -470,7 +472,8 @@ export const fileIcons: FileIcons = {
         'mrf',
       ],
     },
-    { name: 'lib', fileExtensions: ['lib', 'bib'] },
+    { name: 'lib', fileExtensions: ['lib', 'bib', 'a'] },
+    { name: 'dll', fileExtensions: ['dll', 'ilk', 'so'] },
```

Comment:
there is the "so" file extension coming from, what does it mean?

Reply:
`.so` is "Shared Object", and it's (roughly) the Unix equivalent to a dll. These and `.ilk`, "Incremental Link", all represent dynamic libraries. They honestly would all work with under the "lib" category, but the distinction felt significant enough to warrant a separate icon